include/mxnet/c_api.h

```diff
@@ -823,6 +823,23 @@ MXNET_DLL int MXNDArrayToDLPack(NDArrayHandle handle,
 */
 MXNET_DLL int MXNDArrayFromDLPack(DLManagedTensorHandle dlpack,
                                   NDArrayHandle *out_handle);
+
+/*!
+* \brief Create a NDArray backed by a dlpack tensor.
+*
+* This allows us to create a NDArray using the memory
+* allocated by an external deep learning framework
+* that is DLPack compatible.
+*
+* The memory is retained until the NDArray went out of scope.
+*
+* \param dlpack the pointer of the input DLManagedTensor
+* \param out_handle pointer holder to get pointer of NDArray
+* \return 0 when success, -1 when failure happens
+*/
+MXNET_DLL int MXNDArrayFromDLManagedTensor(DLManagedTensorHandle dlpack,
```

Comment:
The most important reason that I didn't call the original API is its overly-strong underlying assumption made me scared. Correct me if I was wrong, but I feel that it asks the DLManagedTensor to be alive throughout the entire lifetime of the external object. To be more specific, in the implementation of `NDArray::FromDLPack`, the deleter would assume the argument `tensor` always points to a valid address - this doesn't usually hold, especially for that in the frontend user may delete the numpy's ndarray. I was not aware why it is initially designed like this.

Reply:
I wrote the function FromDLPack, and it is my mistake. Could you please replace FromDLPack with FromDLManagedTensor, but keep the function name FromDLPack? Thank you!

Reply:
@wkcn No worries! I will try. Thanks!

Reply:
I tried several hours unifying these two implementations, but not able to resolve some core dumps in the unittest `test_ndarray.test_dlpack` because `manager_ctx` points to an invalid address (something like 0x6). Maybe we can do it later.

Reply:
Done per discussion with @wkcn. Now we unify `FromDLPack ` and `FromDLManagedTensor `. Thank you so much!